### PR TITLE
Clarify "must be supplied as keyword" wording in coordinate-frame docstrings

### LINKED ISSUE #3374: HeliographicCarrington doesn't error if obstime isn't supplied
[The docs](https://docs.sunpy.org/en/latest/api/sunpy.coordinates.frames.HeliographicCarrington.html) state that:

>... (note that obstime must be supplied as a keyword):

But the following code runs without error:
```python
import astropy.units as u
from sunpy.coordinates import frames

lon = [1, 2] * u.deg
lat = [1, 2] * u.deg
r = [1, 2] * u.au

c = frames.HeliographicCarrington(lon=lon, lat=lat, radius=r)
```

I think that this should error given that `obstime` hasn't been provided to `HeliographicCarrington`.

### PR DESCRIPTION
**Pull request by dstansby**
Fixes #3374

### CODE REVIEW

File: sunpy/coordinates/frames.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ class HeliographicCarrington(HeliographicStonyhurst):
     This system differs from Stonyhurst Heliographic (HGS) in its definition of longitude.
 
     A new instance can be created using the following signatures
-    (note that ``obstime`` must be supplied as a keyword)::
+    (note that if supplied, ``obstime`` must be a keyword argument)::
 
         HeliographicCarrington(lon, lat, obstime)
```
Comment (Cadair) [inline sunpy/coordinates/frames.py]: are these lines a little misleading?

**Comment by ayshih** [comment]
If this wording change is going to be made, it should be made for all frames, not just `HeliographicCarrington`.

**Comment by dstansby** [comment]
I think this is the only location where the wording is a problem; I've fixed the example signatures where kwargs are required.

File: sunpy/coordinates/frames.py [hunk 2]
```diff
@@ -153,9 +153,9 @@ class HeliographicStonyhurst(SunPyBaseCoordinateFrame):
     (note that ``obstime`` and ``representation_type`` must be supplied as
     keywords)::
```
Comment (ayshih) [inline sunpy/coordinates/frames.py]: Match wording here for consistency

File: sunpy/coordinates/frames.py [hunk 3]
```diff
@@ -311,8 +311,8 @@ class Heliocentric(SunPyBaseCoordinateFrame):
...
```
Comment (ayshih) [inline sunpy/coordinates/frames.py]: Match wording here for consistency

File: sunpy/coordinates/frames.py [hunk 4]
```diff
@@ -375,8 +375,8 @@ class Helioprojective(SunPyBaseCoordinateFrame):
     A new instance can be created using the following signatures
     (note that ``obstime`` and ``observer`` must be supplied as keywords)::
```
Comment (ayshih) [inline sunpy/coordinates/frames.py]: Match wording here for consistency

**Comment by nabobalis** [comment]
I think the changes look fine but are you happy with them now @ayshih?

File: sunpy/coordinates/frames.py [hunk 5]
```diff
@@ -308,11 +308,11 @@ class Heliocentric(SunPyBaseCoordinateFrame):
     by 90 degrees compared to the convention of the Heliocentric Radial (HCR) system.
 
     A new instance can be created using the following signatures
-    (note that ``obstime`` and ``representation_type`` must be supplied as
-    keywords)::
+    (note that if supplied ``obstime`` and ``representation_type``, must be
```
Comment (ayshih) [inline sunpy/coordinates/frames.py]:
```suggestion
    (note that if supplied, ``obstime`` and ``representation_type`` must be
```

File: sunpy/coordinates/frames.py [hunk 6]
```diff
@@ -308,11 +308,11 @@ class Heliocentric(SunPyBaseCoordinateFrame):
...
+    keyword arguments)::
 
-        Heliocentric(x, y, z, obstime)
-        Heliocentric(rho, psi, z, obstime, representation_type='cylindrical')
+        Heliocentric(x, y, z, obstime=obstime)
```
Comment (ayshih) [inline sunpy/coordinates/frames.py]:
```suggestion
        Heliocentric(x, y, z, obstime=obstime, observer=observer)
```

File: sunpy/coordinates/frames.py [hunk 7]
```diff
@@ -308,11 +308,11 @@ class Heliocentric(SunPyBaseCoordinateFrame):
...
+        Heliocentric(rho, psi, z, representation_type='cylindrical', obstime=obstime)
```
Comment (ayshih) [inline sunpy/coordinates/frames.py]:
```suggestion
        Heliocentric(rho, psi, z, representation_type='cylindrical', obstime=obstime, observer=observer)
```

**Comment by wtbarnes** [comment]
Looks like this needs a rebase as well to include the fix in #3439. This should fix the tests.

**Pull request merged**